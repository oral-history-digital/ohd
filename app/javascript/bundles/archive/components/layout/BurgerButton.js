import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function BurgerButton({ open, onClick }) {
    return (
        <button type="button"
                className={classNames('BurgerButton', {
                    'is-open': open,
                    'is-closed': !open,
                })}
                onClick={onClick}>
            <i className={classNames('BurgerButton-icon', 'fa', {
                'fa-close': open,
                'fa-bars': !open,
            })}/>
        </button>
    );
}

BurgerButton.propTypes = {
    open: PropTypes.bool.isRequired,
    onClick: PropTypes.func.isRequired,
};

export default BurgerButton;
