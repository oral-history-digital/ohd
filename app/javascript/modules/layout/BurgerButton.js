import PropTypes from 'prop-types';
import classNames from 'classnames';
import { FaTimes, FaBars } from 'react-icons/fa';

function BurgerButton({
    className,
    open,
    onClick,
}) {
    return (
        <button
            type="button"
            className={classNames(className, 'Button', 'BurgerButton', {
                'is-open': open,
                'is-closed': !open,
            })}
            onClick={onClick}
        >
            {
                open ?
                    <FaTimes className="BurgerButton-icon" /> :
                    <FaBars className="BurgerButton-icon" />
            }
        </button>
    );
}

BurgerButton.propTypes = {
    className: PropTypes.string,
    open: PropTypes.bool,
    onClick: PropTypes.func.isRequired,
};

BurgerButton.defaultProps = {
    open: false,
};

export default BurgerButton;
