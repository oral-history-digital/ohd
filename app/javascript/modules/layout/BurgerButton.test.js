import { renderToStaticMarkup } from 'react-dom/server';
import BurgerButton from './BurgerButton';

it('renders correctly in closed state', () => {
    const html = renderToStaticMarkup(<BurgerButton onClick={() => {}} />);
    expect(html).toMatchSnapshot();
});

it('renders correctly in open state', () => {
    const html = renderToStaticMarkup(<BurgerButton open onClick={() => {}} />);
    expect(html).toMatchSnapshot();
});
