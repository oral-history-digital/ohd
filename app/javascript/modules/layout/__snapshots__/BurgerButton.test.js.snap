// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly in closed state 1`] = `
<button
  className="Button BurgerButton is-closed"
  onClick={[Function]}
  type="button"
>
  <svg
    className="BurgerButton-icon"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
    />
  </svg>
</button>
`;

exports[`renders correctly in open state 1`] = `
<button
  className="Button BurgerButton is-open"
  onClick={[Function]}
  type="button"
>
  <svg
    className="BurgerButton-icon"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 352 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
    />
  </svg>
</button>
`;
